### PR TITLE
feat(visualizations): Improving shareReplay visualization

### PR DESCRIPTION
**Pull request by cpyle0819**
* Add console logs to demonstrate 'share' portion of the operator.
* Change the timing of the outputs a bit to visualize 'share' better.
* Remove comment about improvements being needed
* Do some renaming.

sharereplay

### CODE REVIEW

**Comment by cpyle0819** [comment]
Closed in favor of #23

**Comment by ajcrites** [comment]
In order for this to show off the share capabilities it probably needs some sort of console log to indicate that it's only subscribing once (other than the  `<marbles>` subscription).

Otherwise it's all right, but I'd prefer more thoughtful names ... `delayedOutput` or something like that.

**Comment by cpyle0819** [comment]
I can rename the variables. Doesn't the fact that the 'replay' happens all at once show off the share capabilities?

**Comment by ajcrites** [comment]
Another issue that's really just a consequence of the overlapping marbles of the visualization is that it's hard to tell that only `3, 4` are replayed for the delayed output. This should be pointed out if it's not already.

**Comment by cpyle0819** [comment]
> Another issue that's really just a consequence of the overlapping marbles of the visualization is that it's hard to tell that only `3, 4` are replayed for the delayed output. This should be pointed out if it's not already.

I was thinking about the over lap. I wonder how it would look to have a hover effect on a marble where it 'expands' vertically all the marbles that are overlapping

**Comment by ajcrites** [comment]
> Doesn't the fact that the 'replay' happens all at once show off the share capabilities?

This shows off the replay capabilities, not the share capabilities. The point of share is to be able to reuse a source without resubscribing to the original source.

**Comment by cpyle0819** [comment]
Did some renaming. Added console logs. Removed the 4th marble I had added as it seemed unnecessary upon reflection.

File: src/app/visualizations/operators/shareReplay.ts [hunk 1]
```diff
@@ -12,33 +12,37 @@ import { take, shareReplay, mergeMapTo } from 'rxjs/operators';
...
+      <code>delayedOutput</code>, since it's subscribing to our subject, does
+      not output a log.
+    </p>
+    <p>
+      The replayed values are emitted all at once. It's hard to see because the
+      marbles overlap. The only difference from
       <a routerLink="/share"><code>share</code></a> is that
       <code>shareReplay</code> can replay values. <code>share</code> does not
-      replay any values. <code>shareReplay()</code> will replay
+      replay any values. <code>shareReplay</code> will replay
```
Comment (ajcrites) [inline src/app/visualizations/operators/shareReplay.ts:29]: The parentheses were intentional here to indicate a call to `shareReplay` with no arguments.

File: src/app/visualizations/operators/shareReplay.ts [hunk 2]
```diff
@@ -12,33 +12,37 @@ import { take, shareReplay, mergeMapTo } from 'rxjs/operators';
...
-
-  input = timer(0, 1000).pipe(take(4));
+  input = timer(0, 1000).pipe(
+    take(4),
+    // tslint:disable-next-line:no-console ... open your console!
+    tap(val => val === 0 && console.log('new producer created!')),
+  );
   subject = this.input.pipe(shareReplay(2));
-  output = timer(3500).pipe(mergeMapTo(this.subject));
+  delayedOutput = timer(2000).pipe(mergeMapTo(this.subject));
```
Comment (ajcrites) [inline src/app/visualizations/operators/shareReplay.ts:47]: Is this enough time to limit what is replayed?

**Review by ajcrites — CHANGES_REQUESTED** [review]

**Pull request closed without merging**